apps/total-typescript/src/components/exercise-overlay.tsx

```diff
@@ -164,6 +166,7 @@ const DefaultOverlay: React.FC<OverlayProps> = ({handlePlay}) => {
   const {nextExercise, module, path, exercise} = useMuxPlayer()
   const router = useRouter()
   const {image} = module
+  const addProgressMutation = trpc.useMutation(['progress.add'])
```

Comment:
this has `status` `data` properties so you can show indicators and other nice bits while it's mutating.

slick af. great. 😱🤩